tests/test_client_context.py

```
@@ -213,3 +216,204 @@ def test_render_payload_is_valid_json_inside_delimiters():
+def test_compute_patch_no_facts_returns_none_facts_patch():
+    # No ``facts`` in the push => facts_patch is None => the SQL leaves the
+    # _client_context namespace untouched (so a state-only replace can't wipe
+    # facts). This is the load-bearing distinction from an empty {} facts.
+    _, fp, _, _, fk = compute_context_patch(
+        {}, state={"cart_id": "c"}, facts=None, merge="shallow", config=_cfg()
+    )
+    assert fp is None and fk == []
+
+
+def test_compute_patch_replace_nulls_cleared_state_keys():
+    sp, fp, replace, sk, fk = compute_context_patch(
```

Ruff (0.15.17)

[warning] Unpacked variable `sk` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)